clusterloader2/pkg/measurement/common/api_availability_measurement.go

```diff
@@ -48,126 +48,208 @@ func createAPIAvailabilityMeasurement() measurement.Measurement {
 
 type apiAvailabilityMeasurement struct {
 	isRunning           bool
+	isPaused            bool
+	pauseCh             chan struct{}
+	unpauseCh           chan struct{}
 	stopCh              chan struct{}
-	hosts               []string
+	pollFrequency       time.Duration
+	hostIPs             []string
 	summaries           []measurement.Summary
 	clusterLevelMetrics *apiAvailabilityMetrics
-	hostLevelMetrics    map[string]*apiAvailabilityMetrics
-	wg                  sync.WaitGroup
+	// Metrics per host internal IP.
+	hostLevelMetrics map[string]*apiAvailabilityMetrics
+	wg               sync.WaitGroup
+	lock             sync.Mutex
 }
 
-func (a *apiAvailabilityMeasurement) updateMasterAvailabilityMetrics(c clientset.Interface, config *measurement.Config, provider provider.Provider) {
-	for _, host := range a.hosts {
-		// SSH and check the health of the host
-		command := fmt.Sprintf("curl -f -s -k %slocalhost:%v/healthz", "https://", 443)
-		sshResult, err := measurementutil.SSH(command, host+":22", provider)
-		availability := err != nil || sshResult.Code != 0
-		a.updateAvailabilityMetrics(availability, a.hostLevelMetrics[host])
+func (a *apiAvailabilityMeasurement) updateHostAvailabilityMetrics(c clientset.Interface, provider provider.Provider) {
+	for _, ip := range a.hostIPs {
+		statusCode, err := a.pingHost(ip)
+		availability := statusCode == strconv.Itoa(http.StatusOK)
+		if err != nil {
+			klog.Warningf("execservice issue: %s", err.Error())
+		}
+		if !availability {
+			klog.Warningf("host %s not available", ip)
+		}
+		a.hostLevelMetrics[ip].update(availability)
+	}
+}
+
+func (a *apiAvailabilityMeasurement) pingHost(hostIP string) (string, error) {
+	pod, err := execservice.GetPod()
+	if err != nil {
+		return "", fmt.Errorf("problem with GetPod(): %w", err)
 	}
+	cmd := fmt.Sprintf("curl --connect-timeout 5 -s -k -w \"%%{http_code}\" -o /dev/null https://%s:443/readyz", hostIP)
+	output, err := execservice.RunCommand(pod, cmd)
+	if err != nil {
+		return "", fmt.Errorf("problem with RunCommand(): %w", err)
+	}
+	return output, nil
 }
 
 func (a *apiAvailabilityMeasurement) updateClusterAvailabilityMetrics(c clientset.Interface) {
-	// Check the availability of the cluster by issuing a REST call to /healthz end point
-	result := c.CoreV1().RESTClient().Get().AbsPath("/healthz").Do(context.TODO())
+	result := c.CoreV1().RESTClient().Get().AbsPath("/readyz").Do(context.TODO())
```

Comment:
NIT: Not sure if context.TODO is a good choice here. I don't think we have plans to bring the context into measurements, so more appropriate would be context.Background()

Reply:
Done.